keyboards/m65/keymaps/default/keymap.c

```diff
@@ -0,0 +1,418 @@
+#include "m65.h"
+
+#define TAP(keycode) tap_code16(keycode)
+
+enum custom_keycodes {
+  F_Q = SAFE_RANGE, /* do */
+  F_W , /* write */
+  F_E , /* exp */
+  F_R , /* read */
+  F_T , /* */
+  F_Y , /* */
+  F_U , /* */
+  F_I , /* */
+  F_O , /* */
+  F_P , /* */
+  F_A , /* */
+  F_S , /* */
+  F_D , /* */
+  F_F , /* */
+  F_G , /* */
+  F_H , /* */
+  F_J , /* */
+  F_K , /* */
+  F_L , /* */
+  F_Z , /* */
+  F_X , /* */
+  F_C , /* */
+  F_V , /* */
+  F_B , /* */
+  F_N , /* */
+  F_M , /* */
+  C_Q , /* do */
+  C_W , /* write */
+  C_E , /* exp */
+  C_R , /* read */
+  C_T , /* */
+  C_Y , /* */
+  C_U , /* */
+  C_I , /* */
+  C_O , /* */
+  C_P , /* */
+  C_A , /* */
+  C_S , /* */
+  C_D , /* */
+  C_F , /* */
+  C_G , /* */
+  C_H , /* */
+  C_J , /* */
+  C_K , /* */
+  C_L , /* */
+  C_Z , /* */
+  C_X , /* */
+  C_C , /* */
+  C_V , /* */
+  C_B , /* */
+  C_N , /* */
+  C_M , /* */
```

Comment:
I'm not convinced you need to define all of these extra keycodes. I suspect you can get away with something like the following in `process_record_user()`:
```
if (!record->event.pressed) return true; /* not entirely sure you should always be returning `true` here */
if (IS_LAYER_ON(_Fortran)) {
    switch (keycode) {
        case KC_G:
            SEND_STRING("https://en.wikipedia.org/wiki/Goto");
            break;
    }
} else if (IS_LAYER_ON(_CPP)) {
    switch (keycode) {
        case KC_G:
              SEND_STRING("https://en.wikipedia.org/wiki/Goto");
              break;
    }
}
```
then all of the `F_*`s and `C_*`s in your keymap can just be `KC_TRNS`.

Reply:
(That said, if you really must add extra keycodes, you may as well rename them from e.g. `F_G` to something more descriptive like `FOR_GOTO` say.)

Reply:
beautiful I will do that